src/main/java/core/net/OnlineWorker.java

```diff
@@ -270,23 +269,23 @@ public static boolean downloadMatchData(MatchKurzInfo info, boolean refresh)
 		showWaitInformation(1);
 		// Only download if not present in the database, or if refresh is true
 		if (refresh || !DBManager.instance().isMatchInDB(matchID)
-				|| DBManager.instance().hasUnsureWeatherForecast(matchID)
-				|| !DBManager.instance().isMatchLineupInDB(SourceSystem.HATTRICK.getValue(), matchID)
+				|| (info.getMatchStatus() != MatchKurzInfo.FINISHED && DBManager.instance().hasUnsureWeatherForecast(matchID))
+				|| !DBManager.instance().isMatchLineupInDB(info.getMatchType().getSourceSystem().getValue(), matchID)
```

Comment:
i think here could be the error: htointegrated matches were checked with wrong SourceSystem (HATTRICK), which led to an additional download of maybe wrong matches.

Reply:
as you can see in my various screenshot above the source system was correctly set to hto integrated